### PR TITLE
Worst PM Nodes

### PR DESCRIPTION
**Pull request by gertie-sheshe**
## Description

Display top 5 worst nodes of PM1, PM2.5 and PM 10 readings. Dropdown to select PM reading.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots
Screenshot 2021-06-25 at 08 59 56

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: public/js/charts/worstPMGraph.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    // loop through the values and sum the total of the P1,2 and 0 values
+    if (
+      value.value_type === 'P2' ||
+      value.value_type === 'P1' ||
+      value.value_type === 'P0'
+    ) {
+      // sum them up
+      sum += Number(value.value);
+    }
+  });
```
Comment (kilemensi) [inline public/js/charts/worstPMGraph.js]: You can't do this; they're different measurements. You have to sum each one individually.

**Review by kilemensi — COMMENTED** [review]
👍🏾

File: public/js/charts/worstPMGraph.js (hunk 1 above)
Comment (gertie-sheshe) [reply public/js/charts/worstPMGraph.js]: ahh, okay. So with that in mind then a node should return 3 averages? For P1, P2 and P0? @kilemensi

Comment (kilemensi) [reply public/js/charts/worstPMGraph.js]: Yes!

Comment (gertie-sheshe) [reply public/js/charts/worstPMGraph.js]: Great, so if a node is returning 3 averages, then what's to be displayed on the chart?  The worst of P1? The worst of P2?  The worst of the three? @kilemensi

Comment (kilemensi) [reply public/js/charts/worstPMGraph.js]: For simplicity, we can pick one and show that @gertie-sheshe ... e.g. P2 only.

But if possible, you can show each individual measurement as we did in the [Data4SDGs](https://data4sdgs.sensors.africa/dashboard)

**Review by nerdlet — APPROVED** [review]
Looks good! A quick reminder to fill in details in the PR description

File: public/dashboard/index.html [hunk 2]
```diff
@@ -297,7 +273,11 @@ <h4 class="footer-heading">FUNDED BY</h4>
   <script src="/js/sheets.js" type="text/javascript"></script>
   <script src="/js/aq.js" type="text/javascript"></script>
   <script src="/js/charts/countyCoverage.js"></script>
+<<<<<<< HEAD
+  <script src="/js/charts/worstPMGraph.js"></script>
+=======
   <script src="/js/charts/sensorCoverage.js"></script>
+>>>>>>> master
```
Comment (kilemensi) [inline public/dashboard/index.html]: ?

File: public/js/charts/worstPMGraph.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+const ctx = document.getElementById('myChart').getContext('2d');
+
+const renderWorstNodesChart = (labels, data) => {
+  window.aq.charts.worstNodes.el = new window.Chart(ctx, {
+    type: 'bar',
+    data: {
+      labels,
+      datasets: [
+        {
+          label: '# of Sensors',
```
Comment (kilemensi) [inline public/js/charts/worstPMGraph.js]: Can't be # of nodes.

File: public/js/charts/worstPMGraph.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+async function worstPMNodes() {
+  const nodes = await window.aq.getNodes(); // get the nodes
+
+  const resultAverages = nodes.map((data) => {
+    // store the no of sensors in a node
+    const averageResults = returnPMAverage(data.sensors);
+
+    if (averageResults.length > 0) {
+      // do an average here
+    }
```
Comment (kilemensi) [inline public/js/charts/worstPMGraph.js]: ?

File: public/dashboard/index.html (hunk 2 above)
Comment (gertie-sheshe) [reply public/dashboard/index.html]: Removed

File: public/js/charts/worstPMGraph.js (hunk 3 above)
Comment (gertie-sheshe) [reply public/js/charts/worstPMGraph.js]: Hehe. Changed

File: public/js/charts/worstPMGraph.js (hunk 4 above)
Comment (gertie-sheshe) [reply public/js/charts/worstPMGraph.js]: This is an older state. It's updated @kilemensi

**Review by kilemensi — COMMENTED** [review]
👍🏾 

P0, P1, P2 are supposed to be different readings but I don't see them changing here
Peek 2021-06-25 09-59

**Comment by gertie-sheshe** [comment]
> 👍🏾
> 
> P0, P1, P2 are supposed to be different readings but I don't see them changing here
> Peek 2021-06-25 09-59

Fixed that.
worstnodes

**Comment by kilemensi** [comment]
Much better @gertie-sheshe 

1. Seems like there is a glitch on hover:
Peek 2021-06-25 09-59
2. Node numbers should start at 1
3. The top five nodes that measured the highest (worst) levels of ~PM1, PM 2.5 and PM10~PM over the past ~four~one week. 

Bonus
=====
May be we should show full range of possible values on Y axis so that someone is able to see if they're indeed in toxic range or not.
pm_levels

**Pull request merged**